update/Pos/Update/Worker.hs

```diff
@@ -40,11 +40,15 @@ usWorkers = (map fst [processNewSlotWorker, checkForUpdateWorker], mempty)
               NewSlotTerminationPolicy "Update.processNewSlot"
         }
     processNewSlotWorker =
+        -- TODO: [CSL-198] Catch and report non-critical errors, as
+        -- 'onNewSlot' no longer swallows all exception and rethrows them instead.
```

Comment:
I don't see why this worker may fail in normal cases, so let's stop program if it does (i. e. not catch anything here).